### PR TITLE
Nightvision for Combat Extended

### PR DESCRIPTION
**Pull request by kbatbouta**
## Additions

- Implemented 2 new stats `ApparelNightVisionEfficiency` and `NightVisionEfficiency`.
- Implemented 2 new `StatPart_ApparelStatMaxima` and `StatPart_ApparelStatMinima`.
- Added the required Xml patches for these to work properly.

## Changes

- Now shooting in dark will take the distance into consideration example: targets more than 30 cells should be very hard to hit without night vision and with a unskilled pawn.
- Now shooting will take into account the darkness at the target side.
- Now aiming lead will be less accurate in the dark and behind smoke.

### CODE REVIEW

**Review by N7Huntsman — APPROVED** [review]
Works as intended during test. I'll make some additional xml changes once it's merged, including adding proper stat descriptions and adding night vision stats to some other items that need it.

File: Defs/Stats/Stats_Apparel.xml [hunk 1]
```diff
@@ -32,6 +32,30 @@
...
+    <description>Blah Blah Blah</description>
+    <category>Apparel</category>
+    <minValue>0</minValue>
+    <maxValue>1.0</maxValue>    
+    <hideAtValue>0</hideAtValue>    
+    <defaultBaseValue>0</defaultBaseValue>
+    <toStringStyle>PercentZero</toStringStyle>
+    <showIfUndefined>false</showIfUndefined>   
+    <parts>
+      <li Class="StatPart_Quality">
```
Comment (N7Huntsman) [inline Defs/Stats/Stats_Apparel.xml:48]: I'd rebalance this, such that normal = 1 and go from there.

Comment (kbatbouta) [reply Defs/Stats/Stats_Apparel.xml:48]: Nope, this is intended to force the player to craft goggles instead of looting them.

**Comment by Breadbox2k19** [comment]
I don't think this should be merged hastily, not before AI raids can handle itself at night at least.

**Comment by kbatbouta** [comment]
> I don't think this should be merged hastily, not before AI raids can handle itself at night at least.

I'm not a one man army. I did this based on your stated specifications.

File: Source/CombatExtended/CombatExtended/CE_Utility.cs [hunk 2]
```diff
@@ -555,5 +560,34 @@ public static bool TryGetPrimaryCEWeapon(this Pawn pawn, out Thing gun)
         }
 
         #endregion
+
+        #region Lighting
+
+        /// <summary>
+        /// Used to get the lighting penalty multiplier for a
```
Comment (perkinslr) [inline Source/CombatExtended/CombatExtended/CE_Utility.cs:567]: "for a"... what?

Comment (kbatbouta) [reply Source/CombatExtended/CombatExtended/CE_Utility.cs:567]: lol, for a given range

**Review by perkinslr — APPROVED** [review]

**Review by N7Huntsman — APPROVED** [review]

**Pull request merged**